kong/pdk/response.lua

```diff
@@ -760,6 +821,196 @@ local function new(self, major_version)
+              quality = tonumber(m[1])
+            else
+              m = ngx.re.match(entry, [[^\s*level=(\d)\s*$]], "ajoi")
+              if m then
+                level = tonumber(m[1])
+              end
+            end
+          end
+        end
+
+        if level <= hi_level and quality > hi_quality then
+          if find(name, CONTENT_TYPE_JSON, nil, true) == 1 then
+            type = CONTENT_TYPE_JSON
+
+          elseif find(name, CONTENT_TYPE_GRPC, nil, true) == 1 then
+            type = CONTENT_TYPE_GRPC
+
+          elseif find(name, CONTENT_TYPE_HTML, nil, true) == 1 then
+            type = CONTENT_TYPE_HTML
+
+          elseif find(name, CONTENT_TYPE_XML, nil, true) == 1 then
+            type = CONTENT_TYPE_XML
+
+          elseif find(name, CONTENT_TYPE_PLAIN, nil, true) == 1 then
+            type = CONTENT_TYPE_PLAIN
+          end
+
+          hi_level = level
+          hi_quality = quality
+
+        end
+      end
+
+    end
+
+    return type
+  end
+
+
+  ---
+  -- This function interrupts the current processing and produces an error
+  -- response.
+  --
+  -- It is recommended to use this function in conjunction with the `return`
+  -- operator, to better reflect its meaning:
+  --
+  -- ```lua
+  -- return kong.response.exit(500, "Error", {["Content-Type"] = "text/html"})
+  -- ```
+  --
+  -- The first argument `status` will set the status code of the response that
+  -- will be seen by the client. The status code must be of an error, i.e.
+  -- >399.
+  --
+  -- The second, optional, `message` argument will set the message describing
+  -- the error, which will be written in the body.
+  --
+  -- The third, optional, `headers` argument can be a table specifying response
+  -- headers to send. If specified, its behavior is similar to
+  -- `kong.response.set_headers()`.
+  --
+  -- This method will send the response formatted in JSON, XML, HTML or plain
+  -- text. The actual format is chosen using one of the following options:
+  -- - Manually specifying in `headers` argument using the `Content-Type`
+  --   header.
+  -- - Conform to the `Accept` header from the request.
+  -- - If none of the above is found, fallback to JSON format.
+  -- Content-Length header in the produced response for convenience.
+  -- @function kong.response.error
+  -- @phases rewrite, access, admin_api, header_filter (only if `body` is nil)
+  -- @tparam number status The status to be used (>399)
+  -- @tparam[opt] string message The error message to be used
+  -- @tparam[opt] table headers The headers to be used
+  -- @return Nothing; throws an error on invalid input.
+  -- @usage
+  -- return kong.response.error(403, "Access Forbidden", {
+  --   ["Content-Type"] = "text/plain",
+  --   ["WWW-Authenticate"] = "Basic"
+  -- })
+  --
+  -- ---
+  --
+  -- return kong.response.exit(403, "Access Forbidden")
+  --
+  -- ---
+  --
+  -- return kong.response.exit(403)
+  function _RESPONSE.error(status, message, headers)
```

Comment:
Is there specific reason this couldn't just call `response.exit` like the `error_handlers` did?

Reply:
I wasn't really certain that this was needed, but I chose to do this by: there are errors that can't use templates, like some sent by jwt. And I think that introducing the `Accept`/`Content-Type` treatment could inadvertently change the behavior of some plugins (although that would be correct). 

It's simple to join the behavior of `exit` and `error`, what do you think?